<?xml version="1.0"?>
<!--***********************************************************
	Launches all monitors
	Emily Sheetz, NSTGRO VTE 2023
************************************************************-->
<launch>
	
	<!-- ***** JOINT STATE MONITOR ***** -->

	<!-- parameters for specifying joint state topic -->
	<arg name="joint_state_topic" default="/ihmc_ros/valkyrie/output/joint_states"/>

	<node pkg="val_soft_estop_monitor" type="joint_state_monitor_node" name="ValkyrieJointStateMonitorNode" output="screen">
		<param name="joint_state_topic" type="string" value="$(arg joint_state_topic)"/>
		<!--<param name="" type="" value=""/> -->
	</node>

	<!-- ***** JOINT STATE DELTA MONITOR ***** -->
	<node pkg="val_soft_estop_monitor" type="joint_state_delta_monitor_node" name="ValkyrieJointStateDeltaMonitorNode" output="screen">
		<param name="joint_state_topic" type="string" value="$(arg joint_state_topic)"/>
		<!--<param name="" type="" value=""/> -->
	</node>

	<!-- ***** END-EFFECTOR STREAMING MONITOR ***** -->

	<!-- parameters for specifying end-effector streaming topic -->
	<arg name="ee_streaming_topic" default="/ihmc/valkyrie/toolbox/ik_streaming/input/kinematics_streaming_toolbox_input"/>

	<node pkg="val_soft_estop_monitor" type="end_effector_streaming_monitor_node" name="ValkyrieEndEffectorStreamingMonitorNode" output="screen">
		<param name="ee_streaming_topic" type="string" value="$(arg ee_streaming_topic)"/>
		<!--<param name="" type="" value=""/> -->
	</node>

	<!-- ***** RQT_RECONFIGURE ***** -->

	<!-- rosrun rqt_reconfigure rqt_reconfigure -->
	<node pkg="rqt_reconfigure" name="rqt_reconfigure_monitors" type="rqt_reconfigure" output="screen"/>

</launch>
